x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ApplicationPrivilege.java

```diff
@@ -162,10 +162,17 @@ public static void validatePrivilegeName(String name) {
         }
     }
 
-    private static boolean isValidPrivilegeName(String name) {
+    public static boolean isValidPrivilegeName(String name) {
         return VALID_NAME.matcher(name).matches();
     }
 
+    public static void validateActionName(String action) {
```

Comment:
Does it make sense for a new `validateActionName` method to call a new `isValidActionName` method?

For example, there is an existing `validatePrivilegeName` method which calls `isValidPrivilegeName`. Does it make sense to follow that pattern?

In other words, I found it confusing to have a method called `validateActionName` which does validation on action name, but then also calls through to `validatePrivilegeOrActionName`.

Reply:
> Does it make sense for a new validateActionName method to call a new isValidActionName method?

I am unsure what that would achieve.
The `validateActionName` throws two different kinds of errors at the moment. How can the proposal, as I understand it, achieve the same thing when just calling `isValidPrivilegeName`?

I have pushed dea850d34b7d04d1e338001 . Not sure if this helps or not.